geozero/src/wkt/mod.rs

```diff
@@ -40,18 +61,40 @@ pub(crate) mod conversion {
 mod wkb {
     use crate::error::Result;
     use crate::wkb::{FromWkb, WkbDialect};
-    use crate::wkt::{WktString, WktWriter};
+    use crate::wkt::{EwktString, WktDialect, WktString, WktWriter};
+    use crate::CoordDimensions;
     use std::io::Read;
 
     impl FromWkb for WktString {
         fn from_wkb<R: Read>(rdr: &mut R, dialect: WkbDialect) -> Result<Self> {
             let mut out: Vec<u8> = Vec::new();
-            let mut writer = WktWriter::new(&mut out);
+            let mut writer = WktWriter::with_dialect(&mut out, WktDialect::Wkt);
             crate::wkb::process_wkb_type_geom(rdr, &mut writer, dialect)?;
             let wkt = String::from_utf8(out).map_err(|_| {
                 crate::error::GeozeroError::Geometry("Invalid UTF-8 encoding".to_string())
             })?;
             Ok(WktString(wkt))
         }
     }
+
+    impl FromWkb for EwktString {
+        fn from_wkb<R: Read>(rdr: &mut R, dialect: WkbDialect) -> Result<Self> {
+            let mut out: Vec<u8> = Vec::new();
+            let mut writer = WktWriter::with_dialect(&mut out, WktDialect::Ewkt);
+            writer.dims = CoordDimensions::xyzm();
+            crate::wkb::process_wkb_type_geom(rdr, &mut writer, dialect)?;
```

Comment:
not sure if actionable, just checking if this is correct: both the writer and the processor need a dialect?   And when reading, the processor cannot auto-detect the dialect, or can it?

Reply:
So an `Ewkt` string can be parsed from several `Wkb` dialects, which is why we need to pass it as a parameter. In the current implementation, no detection of the `Wkb` dialect is implemented; I guess it would be pretty tricky to do so, as they all have their own quirks.